scripts/daqconf_viewer

```diff
@@ -0,0 +1,261 @@
+
+    async def get_json(self, conf, ver) -> None:
+        self.confname = conf
+        self.version = ver
+        if self.confname != None and self.version != None:
+            async with httpx.AsyncClient() as client:
+                payload = {'name': self.confname, 'version': self.version}
+                r = await client.get(f'{self.hostname}/retrieveVersion', auth=auth, params=payload, timeout=60)
+            self.confdata = r.json()
+
+    def json_into_tree(cls, node, json_data):
+        """Takes a JSON, and puts it into the tree."""
+        from rich.highlighter import ReprHighlighter
+
+        highlighter = ReprHighlighter()
+
+        def add_node(name, node, data) -> None:
+            """Adds a node to the tree.
+            Args:
+                name (str): Name of the node.
+                node (TreeNode): Parent node.
+                data (object): Data associated with the node.
+            """
+            if isinstance(data, dict):
+                node.set_label(Text(f"{{}} {name}"))
+                for key, value in data.items():
+                    new_node = node.add("")
+                    add_node(key, new_node, value)
+            elif isinstance(data, list):
+                node.set_label(Text(f"[] {name}"))
+                for index, value in enumerate(data):
+                    new_node = node.add("")
+                    add_node(str(index), new_node, value)
+            else:
+                node.allow_expand = False
+                if name:
+                    label = Text.assemble(
+                        Text.from_markup(f"[b]{name}[/b]="), highlighter(repr(data))
+                    )
+                else:
+                    label = Text(repr(data))
+                node.set_label(label)
+
+        add_node("", node, json_data)
+
+    def watch_confdata(self, confdata:dict) -> None:
+        if confdata:
+            tree = self.query_one(Tree)
+            tree.clear()
+            self.json_into_tree(tree.root, confdata)
+            tree.root.expand()
+
+    def on_button_pressed(self, event: Button.Pressed) -> None:
+        if self.confdata:
+            self.app.mount(DiffScreen(self.hostname, self.confdata, id='diffscreen'))
+
+class DiffDisplay(Vertical):
+    confdata = reactive(None)
+
+    def __init__(self, hostname, **kwargs):
+        super().__init__(**kwargs)
+        self.hostname = hostname
+        self.confname = None
+        self.version = None
+
+    def compose(self) -> ComposeResult:
+        yield Static(id='diffbox')
+
+    async def get_json(self, conf, ver) -> None:
+        self.confname = conf
+        self.version = ver
+        if self.confname != None and self.version != None:
+            async with httpx.AsyncClient() as client:
+                payload = {'name': self.confname, 'version': self.version}
+                r = await client.get(f'{self.hostname}/retrieveVersion', auth=auth, params=payload, timeout=60)
```

Comment:
I probably would make this timeout 2seconds or so, and then make
```
r = {"error": f"couldn't retrieve the configuration at {self.hostname}/retrieveVersion (payload: {payload}"}
```
and wrap in in a try-except